Enhancement/remove non living tracer cnt

mnlevy1981 opened the pull request:
Remove the `non_living_biomass_ecosys_tracer_cnt` variable (see issue #137).

Testing: verified that the tracer names and metadata did not change with this refactor by running a single POP CECO (+ciso) comparison test on both Cheyenne (intel 17) and Hobart (nag); this testing did catch some initial typos, and one of the errors flagged (I had accidentally changed the `ALK` tracer to `alk`) also caught a typo in a MARBL error traceback.

mnlevy1981 committed:
non_living_tracer_cnt can be local var

marbl_init_tracer_metadata() passes non_living_biomass_ecosys_tracer_cnt to
marbl_init_non_autotroph_tracer_metadata(), where it is an intent out. However,
the variable is only used in the non_autotroph_tracer_metadata() [it is not
used after being returned], so I updated the interface accordingly.

Next commit will look at how to remove it altogether.

mnlevy1981 committed:
Refactor marbl_init_non_autotroph_tracer_metadata

What was previously marbl_init_non_autotroph_tracer_metadata has been renamed
so that tracer is plural: marbl_init_non_autotroph_tracers_metadata; a new
marbl_init_non_autotroph_tracer_metadata routine has been introduced for
changing the metadata one tracer at a time. This removes the need for
non_living_biomass_ecosys_tracer_cnt, which was implicitly requiring that the
non-living biomass tracers had tracer indices 1 - N.

Also fixed a typo in an error traceback.

klindsay28 commented:
If the comparison
`if (n == alk_ind .or. n == alk_alt_co2_ind) then`
in `subroutine marbl_init_non_autotroph_tracer_metadata` is changed to
`if (short_name == 'ALK' .or. short_name == 'ALK_ALT_CO2') then`
then the arguments `marbl_tracer_indices` and `n` can both be removed from the subroutine.

At that point, I call into question the need for (and usefulness of) the `associate` construct in `subroutine marbl_init_non_autotroph_tracers_metadata`. Each individual tracer index is only used once, and as you get deeper into the sequence of calls to `marbl_init_non_autotroph_tracer_metadata`,  you separate the usage of the tracer index from its association into `marbl_tracer_indices`.

mnlevy1981 committed:
Clean up marbl_init_non_autotroph_tracer_metadata

Removed a couple of un-necessary arguments from the subroutine interface, which
also let me remove an associate statement in
marbl_init_non_autotroph_tracers_metadata.
